src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1541,6 +1543,60 @@ def reoffset(o):
 
         minimize_int(offset, reoffset)
 
+    def shrink_offset_pairs(self):
+        """Lower any two blocks offset from each other the same ammount.
+
+        Try lowering every non-zero pair of non-shrinking blocks by a
+        common offset.
+
+        """
+        self.debug('Shrinking offset pairs.')
+
+        current = self.shrink_target
+        blocked = [current.buffer[u:v] for u, v in current.blocks]
+
+        n = len(blocked)
+
+        shrink_blocks = [b for b in range(n) if self.is_shrinking_block(b)]
+
+        ints = [int_from_bytes(blocked[i]) for i in range(n)]
+
+        nonzero = sorted([b for b in range(n)
+                          if ints[b] > 0 and b not in shrink_blocks])
+
+        if len(nonzero) == 0:
+            return
+
+        # TODO: Should we try combinations of > 2?
+        pairs = itertools.combinations(nonzero, 2)
+
+        # Try reoffseting every pair
+        def reoffset_pair(pair, o):
+            m = min([ints[p] for p in pair])
+            new_blocks = list(blocked)
+            for i in pair:
+                new_blocks[i] = int_to_bytes(ints[i] + o - m, len(blocked[i]))
+
+            buffer = hbytes().join(new_blocks)
+            if sort_key(buffer) < sort_key(self.shrink_target.buffer):
```

Comment:
The coverage job failed because this condition is always true.

More importantly, this function isn't checking if the buffer actually works (that is, makes the test exhibit failure) - see `return self.incorporate_new_buffer(hbytes().join(new_blocks))` above?  That `minimize_int` call is actually doing a lot of state mutation!  We want to try each candidate buffer that the pass thinks might work, instead of only one at the end of the pass.

In turn, this will change the style of the loop - see ['weird loops' in our internals guide](https://github.com/HypothesisWorks/hypothesis-python/blob/master/guides/internals.rst#weird-loops) so that we can make some progress on each pass without the big-O being worse than it needs to be.  

This will *also* interact in complicated ways with *n*-ary combinations (probably a good idea!), so I'd suggest getting some input from @DRMacIver specifically on this before sinking too much time into it.

Reply:
Yeah, this should probably be `if self.incorporate_new_buffer(buffer):`